# HG changeset patch
# User Rik <rik@octave.org>
# Date 1478562526 28800
#      Mon Nov 07 15:48:46 2016 -0800
# Node ID 44d9d526d1e3d02d7a834cc2b9ee4538ac293907
# Parent  6f111a0f32c3f7bd7d577eec07539b04f6dc040d
importdata.m: Allow importing of sound files via audioread.

* importdata.m: Remove special code for 'wav' files that used wavread.
Replace call to error() with call to audioread() for input file extensions
that are sound files.

diff --git a/scripts/io/importdata.m b/scripts/io/importdata.m
--- a/scripts/io/importdata.m
+++ b/scripts/io/importdata.m
@@ -92,18 +92,18 @@ function [output, delimiter, header_rows
   endif
 
   ## Check file format
   ## Get the extension from the filename.
   [~, ~, ext, ~] = fileparts (fname);
   ext = lower (ext);
 
   switch (ext)
-    case {".au", ".snd", ".flac", ".ogg"}
-      error ("importdata: not implemented for file format %s", ext);
+    case {".au", ".snd", ".flac", ".ogg", ".wav", ".wave"}
+      [output.data, output.fs] = audioread (fname);
     case {".avi", ".mj2", ".mpg", ".asf", ".asx", ".wmv", ".mp4", ".m4v", ...
           ".mov"}
       error ("importdata: not implemented for file format %s", ext);
     case {".bmp", ".cur", ".gif", ".hdf", ".ico", ".jpe", ".jpeg", ".jpg", ...
           ".jp2", ".jpf", ".jpx", ".j2c", ".j2k", ".pbm", ".pcx", ".pgm", ...
           ".png", ".pnm", ".ppm", ".ras", ".tif", ".tiff", ".xwd"}
       delimiter = NaN;
       header_rows = 0;
@@ -118,20 +118,16 @@ function [output, delimiter, header_rows
     case {".ods", ".sxc", ".fods", ".uos", ".xml"}
       ## unimplemented.m only knows ML functions; odsread isn't one but is in OF
       try
         output = odsread (fname);
       catch
         ## Fall back to unimplemented.m.
         output = xlsread (fname);
       end_try_catch
-    case {".wav", ".wave"}
-      delimiter = NaN;
-      header_rows = 0;
-      [output.data, output.fs] = wavread (fname);
     otherwise
       ## Assume the file is in ASCII format.
       [output, delimiter, header_rows] = ...
         importdata_ascii (fname, delimiter, header_rows);
   endswitch
 
   ## If there are any empty fields in the output structure, then remove them
   if (isstruct (output) && numel (output) == 1)
